homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,335 @@
+"""
+Support for GitHub.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.github/
+"""
+from datetime import timedelta
+import json
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_NAME, CONF_ACCESS_TOKEN, CONF_NAME, CONF_PASSWORD, CONF_PATH,
+    CONF_SCAN_INTERVAL, CONF_URL, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['PyGithub==1.43.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GROUP_AUTH = 'authentication'
+CONF_REPOS = 'repositories'
+
+ATTR_BRANCHES = 'branches'
+ATTR_LAST_COMMIT = 'last_commit'
+ATTR_LATEST_RELEASE = 'latest_release'
+ATTR_OPEN_ISSUES = 'open_issues'
+ATTR_OPEN_PULL_REQUESTS = 'open_pull_requests'
+ATTR_PATH = 'path'
+ATTR_STARGAZERS = 'stargazers'
+ATTR_TOPICS = 'topics'
+
+DEFAULT_NAME = 'GitHub'
+
+SCAN_INTERVAL = timedelta(seconds=300)
+
+REPO_SCHEMA = vol.Schema({
+    vol.Required(CONF_PATH): cv.string,
+    vol.Optional(CONF_NAME): cv.string
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Exclusive(CONF_ACCESS_TOKEN, CONF_GROUP_AUTH): cv.string,
+    vol.Exclusive(CONF_USERNAME, CONF_GROUP_AUTH): cv.string,
```

Comment:
For increased security, would it make sense to only support the access token?

Reply:
If both keys are kept, use `cv.has_atleast_one_key` to make sure that one is present.

Reply:
I agree. The username and password will cause issues when using 2fa anyways, which I believe everyone should use. Using an access token, you can lock down access also